test/ejabberd_tests/tests/xep_0352_csi_SUITE.erl

```diff
@@ -0,0 +1,251 @@
+
+    {ok, Alice2, _AliceProps2, _} = escalus_connection:start(AliceSpec),
+
+    escalus_connection:send(Alice2, escalus_stanza:presence(<<"available">>)),
+
+    then_client_receives_message(Alice2, MsgsToAlice),
+
+    ok.
+
+
+alice_gets_buffered_messages_after_stream_resumption(Config) ->
+    ConnSteps = [start_stream,
+                 stream_features,
+                 authenticate,
+                 bind,
+                 session,
+                 stream_resumption],
+    NewConfig = escalus_fresh:create_users(Config, [{alice, 1}, {bob, 1}]),
+    AliceSpec = escalus_users:get_userspec(NewConfig, alice),
+    BobSpec = escalus_users:get_userspec(NewConfig, bob),
+    {ok, Alice0, AliceProps, _} = escalus_connection:start(AliceSpec,
+                                                           ConnSteps),
+    JID = make_jid_from_spec(AliceProps),
+    Alice = Alice0#client{jid = JID},
+
+    escalus_connection:send(Alice, escalus_stanza:presence(<<"available">>)),
+    escalus:wait_for_stanza(Alice),
+    {ok, Bob, _BobProps, _} = escalus_connection:start(BobSpec),
+
+    given_client_is_inactive(Alice),
+
+    MsgsToAlice = given_client_is_inactive_and_messages_sent(Alice, Bob, 5),
+
+    %% then Alice disconnects
+
+    escalus_connection:kill(Alice),
+
+    SMID = proplists:get_value(smid, AliceProps),
+    ResumeSession = [start_stream,
+                     stream_features,
+                     authenticate,
+                     mk_resume_stream(SMID, 1)],
+
+    {ok, Alice2, _AliceProps2, _} = escalus_connection:start(AliceSpec,
+                                                             ResumeSession),
+
+    escalus_connection:send(Alice2, escalus_stanza:presence(<<"available">>)),
+
+    then_client_receives_message(Alice2, MsgsToAlice),
+
+    ok.
+
+make_jid_from_spec(AliceProps) ->
+    AliceUsername = proplists:get_value(username, AliceProps),
+    AliceServer = proplists:get_value(server, AliceProps),
+    <<AliceUsername/binary, "@", AliceServer/binary>>.
+
+mk_resume_stream(SMID, PrevH) ->
+    fun (Conn, Props, Features) ->
+            escalus_connection:send(Conn, escalus_stanza:resume(SMID, PrevH)),
+            Resumed = escalus_connection:get_stanza(Conn, get_resumed),
+            true = escalus_pred:is_sm_resumed(SMID, Resumed),
+            {Conn, [{smid, SMID} | Props], Features}
+    end.
+alice_gets_message_after_buffer_overflow(Config) ->
+    escalus:fresh_story(Config, [{alice, 1}, {bob, 1}], fun(Alice, Bob) ->
+        Msgs = given_client_is_inactive_and_messages_sent(Alice, Bob, ?CSI_BUFFER_MAX+5),
+
+        {Flushed, Awaiting} = lists:split(?CSI_BUFFER_MAX+1, Msgs),
+
+        then_client_receives_message(Alice, Flushed),
+        %% and no other stanza
+        escalus_assert:has_no_stanzas(Alice),
+        %% Alice activates
+        escalus:send(Alice, csi_stanza(<<"active">>)),
+        %% ands gets remaining stanzas
+        then_client_receives_message(Alice, Awaiting)
+    end).
+
```

Comment:
...and here. There are also some other double empty lines that seems quite random...